jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -677,13 +677,16 @@ ecma_builtin_helper_def_prop (ecma_object_t *obj_p, /**< object */
   prop_desc.value = value;
 
   prop_desc.is_writable_defined = true;
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wconversion"
   prop_desc.is_writable = writable;
 
   prop_desc.is_enumerable_defined = true;
   prop_desc.is_enumerable = enumerable;
 
   prop_desc.is_configurable_defined = true;
   prop_desc.is_configurable = configurable;
+#pragma GCC diagnostic pop
```

Comment:
Oh, the CONFIG_C99_BOOL8 definition is relevant for nuttx (and tinyara) system includes only.
For clients (which jerry is one of), this results in:
#ifdef CONFIG_C99_BOOL8
#define bool _Bool
#else
#define bool _Bool8
#endif
(...)
typedef uint8_t _Bool8;

_Bool is standard C99 type.